src/pages/programIndicators/form/OrgUnitField.tsx

```diff
@@ -0,0 +1,191 @@
+    ): DisplayableModel[] => {
+        if (!program.programStages) {
+            return []
+        }
+
+        return program.programStages.flatMap(({ programStageDataElements }) =>
+            programStageDataElements.reduce(
+                (acc: DisplayableModel[], { dataElement }) => {
+                    if (dataElement.valueType === ORG_UNIT_VALUE_TYPE) {
+                        acc.push(dataElement)
+                    }
+                    return acc
+                },
+                []
+            )
+        )
+    }
+
+    const programAttributesForProgram: DisplayableModel[] = useMemo(() => {
+        if (!formValues?.program?.programTrackedEntityAttributes) {
+            return []
+        }
+        return formValues.program.programTrackedEntityAttributes.reduce(
+            (
+                acc: DisplayableModel[],
+                {
+                    trackedEntityAttribute,
+                }: {
+                    trackedEntityAttribute: {
+                        id: string
+                        displayName: string
+                        valueType: string
+                    }
+                }
+            ) => {
+                if (trackedEntityAttribute.valueType === ORG_UNIT_VALUE_TYPE) {
+                    acc.push({
+                        id: trackedEntityAttribute.id,
+                        displayName: trackedEntityAttribute.displayName,
+                    })
+                }
+                return acc
+            },
+            []
+        )
+    }, [formValues.program])
+
+    const getOptionsForSelect = (values: ProgramStagesType[]) => {
+        const dataElements =
+            values && values.length > 0
+                ? extractOrgUnitDataElementsFromProgramStage(values[0])
+                : []
+        if (programType === PROGRAM_TYPE_WITHOUT_REGISTRATION) {
+            return [staticOptions.eventDefault, ...dataElements]
+        }
+
+        if (
+            programType === PROGRAM_TYPE_WITH_REGISTRATION &&
+            analyticsType === ANALYTICS_TYPE_EVENT
+        ) {
+            return [
+                staticOptions.eventDefault,
+                ...programAttributesForProgram,
+                ...dataElements,
+                staticOptions.registration,
+                staticOptions.enrollment,
+                staticOptions.ownerAtStart,
+                staticOptions.ownerAtEnd,
+            ]
+        }
+
+        if (
+            programType === PROGRAM_TYPE_WITH_REGISTRATION &&
+            analyticsType === ANALYTICS_TYPE_ENROLLMENT
+        ) {
+            return [
+                staticOptions.enrollmentDefault,
+                ...programAttributesForProgram,
+                staticOptions.registration,
+                staticOptions.ownerAtStart,
+                staticOptions.ownerAtEnd,
+            ]
+        }
+
+        return []
+    }
+
+    return hasRequiredParams ? (
+        <ModelSingleSelectField
```

Comment:
There seems to be a bug if you have an option selected, and you change the Program or `analyticsType`. A request that doesnt make much sent (eg to /program/EVENT) will be sent (because of some logic in `ModelSingleSelect`).
And the field will be stuck in loading. 

It might be worth to create a custom hook to grab and calculate the options and render a simple `SingleSelect` instead of the workarounds here using `ModelSingleSelect`?

Also maybe changing program or analyticstype should reset the Organisation Unit field?